File: 101-APCS/Contents/b966/b966.py
import sys

for n in sys.stdin:
    n = int(n)

    point = []
    for i in range(n):
        point.append([int(e) for e in input().split()])
    
    point = sorted(point, key=lambda x : (x[0], x[1]))

    count = 0
    j = 0
    while j < n:
        start, end = point[j][0], point[j][1]

        while j + 1 < n and point[j + 1][0] < end:
            if point[j + 1][1] < end:
                j += 1
            else:
                end = point[j + 1][1]
                j += 1
        
        count += end - start
        j += 1
    
    print(count)

File: 101-APCS/Contents/b966/b966.cpp
#include<iostream>
#include<algorithm>

#define MAX 10000

typedef struct point
{
	int head;
	int tail;
}Point;

int cmp(Point, Point);

using namespace std;

int main(void)
{
	int n, start, end;
	Point p[MAX];
	
	while(cin >> n)
	{
		for(int i = 0; i < n; i++)
			cin >> p[i].head >> p[i].tail;
			
		sort(p, p + n, cmp);
		
		int count = 0;
		for(int j = 0; j < n; j++)
		{
			start = p[j].head;
			end = p[j].tail;
			
			while(j + 1 < n && p[j + 1].head < end)
			{
				if(p[j + 1].tail < end)
					j++;
				else
				{
					end = p[j + 1].tail;
					j++;
				}
			}
			count += end - start;
		}
		
		cout << count << endl;	
	}
	
	return 0;
}

int cmp(Point a, Point b)
{
	return a.head == b.head ? a.tail < b.tail : a.head < b.head;
}
